packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -867,12 +875,18 @@ export class GarbageCollector implements IGarbageCollector {
                 }
                 if (newDeletedNodes.length > 0) {
                     // Call container runtime to delete these nodes and add deleted nodes to this.deletedNodes.
+                    const deletedNodes = this.runtime.updateUnusedRoutes(newDeletedNodes);
+                    deletedNodes.forEach((node, index) => {
```

Comment:
This validation can be moved to the runtime right. It can validate that all the nodes / blobs that were asked to delete were deleted.

Reply:
Good point

Reply:
Actually, the assertion should be done here as calling deleteUnusedNodes here should delete all the nodes, when we call this function during sweep, we do not delete all the nodes as some are blob nodes (for now)

Reply:
Going to move this to the validation PR

Reply:
DeletedNodes should deepEquals newDeletedNodes